JahsieAult/homework.md

```diff
@@ -0,0 +1,164 @@
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+>(t - b) > (r - l)
+
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+>if ((t - b) > (r - l)); {
+	var maxCircumference = 3.14 * ((r - l)/2); 
+	} else {
+		var maxCircumference = 3.14 * ((t - b)/2);
+	}
+
+**d)**
+Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+**e)**
+Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position.
+Define four new variables describing the centermost small rectangle.
+(_Hint: one of the many solutions is very similar to the solution of **2b** above._)
+
+---
+
+**4)** (_Difficulty: easyish_)
+
+Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+* Each square has integer coordinates _(R,C)_ describing its row and column.  Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+* Each square has a unique integer number N from 0 to 63.  These numbers run sequentially left-to-right one row at a time, top to bottom.  Therefore the upper-left square has N==0 and the bottom-right has N==63.
+
+**a)**  Given a particular R and C, find the corresponding N.  That is, write an expression for variable N containing variables R and C.
+>var N = (R * 8) + (C);
+
+**b)**  Given N, find R.  Write an expression for R which contains N.
+>var R = (N - C)/8;
+Pretty sure I got this one wrong.
+
+**c)**  Given N, find C.  Write an expression for C which contains N.
+>Var C = N % 8;
+
+**d)**  Assume the squares are colored black and white, with the upper-left square black.
+Write an expression to set a variable _color_ to either 'black' or 'white', describing the square identified by variables R,C, and N.
+(_Hint: If you prefer, you may use a pair of conditional statements instead of a single expression._)
+>if (N % 2 = 0) {
++	var color =  black
++	} else {
++	var color = white}
+
+---
+
+**5)** (_Difficulty: moderate_)
+
+Suppose you represent a fraction (_n/d_) with 2 integer variables: _n_ for the numerator and _d_ for the denominator.
+If _n_ is greater than _d_, the fraction is "improper", but it can be rewritten as a proper fraction.  For example, "7/4" is improper, but it can be rewritten as "1 3/4", which is proper.
+
+Write an expression for a string expressing the proper form of an improper fraction _n/d_.  For example, when _n==7_ and _d==4_, your resulting string should be "1 3/4".  You may assume both _n_ and _d_ are positive integers and _n_ > _d_, but otherwise you should be able to handle any values of _n_ and _d_.
+(_Hint: you'll need the modulo operator _%_, and you'll probably want to create a couple of extra variables._)
```

Comment:
5) for this one, you'll need three parts: the numerator, the denominator, and the whole number. You've already got the denominator in `d`, so just two more to go! For n, you're basically looking for the bit that's left over after division, which sounds a lot like mod, and the whole number is just the bit that's not left over after division.